pennylane/measurements/__init__.py

```diff
@@ -11,8 +11,133 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""
-This module contains measurements supported by PennyLane.
+r"""
+This module contains all the measurements supported by PennyLane.
+
+Description
+-----------
+
+Measurements
+~~~~~~~~~~~~
+The :class:`MeasurementProcess` class serves as a base class for measurements, and is inherited
+from the :class:`SampleMeasurement`, :class:`StateMeasurement` and :class:`MeasurementTransform`
+classes. These classes are subclassed to implement measurements in PennyLane.
+
+* Each :class:`SampleMeasurement` subclass represents a sample-based measurement, which contains a
+  :func:`SampleMeasurement.process_samples` method that processes the sequence of samples generated
+  by the device. See :class:`CountsMP` for an example.
+
+* Each :class:`StateMeasurement` subclass represents a state-based measurement, which contains a
+  :func:`StateMeasurement.process_state` method that processes the quantum state generated by the
+  device. See :class:`StateMP` for an example.
+
+* Each :class:`MeasurementTransform` subclass represents a measurement process that requires
+  the application of a batch transform, which converts the given quantum script into a batch of
+  quantum scripts and executes them using the device. The main difference between a
+  :class:`MeasurementTransform` and a :func:`~pennylane.batch_transform` is that a batch transform is
+  tracked by the gradient transform, while a :class:`MeasurementTransform` process isn't.
+  See :class:`ClassicalShadowMP` for an example.
+
+.. note::
+
+    The :func:`SampleMeasurement.process_samples`, :func:`StateMeasurement.process_state` and
+    :func:`MeasurementTransform.process` functions should always have the same signature.
+
+.. note::
+
+    A measurement process can inherit from both :class:`SampleMeasurement` and
+    :class:`StateMeasurement` classes, defining the needed logic to process either samples or the
+    quantum state. See :class:`VarianceMP` for an example.
+
```

Comment:
I would suggest to put the comment about the functions vs MP class somewhere around here